plugins/catalog-backend/src/service/request/parseEntityTransformParams.ts

```diff
@@ -20,6 +20,22 @@ import lodash from 'lodash';
 import { RecursivePartial } from '../../util/RecursivePartial';
 import { parseStringsParam } from './common';
 
+function getPathArrayAndValue(input: Entity, field: string) {
+  return field.split('.').reduce(
+    ([pathArray, inputSubset], pathPart, index, fieldParts) => {
+      if (inputSubset.hasOwnProperty(pathPart)) {
```

Comment:
I don't think you can do this in a `reduce` easily - before even going into the `if` statements you need to check basically that `typeof inputSubset === 'object'` or similar, and if it isn't, bail out early and return undefined - because there is no point in trying to descend into a null or a number or so. So this should probably be a for..of loop instead over the parts, or similar

Reply:
You could write a test that tries to fetch `spec.type`, and then a test that tries to fetch `spec.type.doesnotexist`

Reply:
Ohh sure. I was using a "hasOwn" before, but that was not supported in node 14. 😞 

I'll add that test.

I had a for of loop originally, but it felt to cumbersome. I'll switch it back if that's preferred.

Reply:
@freben so 
> You could write a test that tries to fetch spec.type, and then a test that tries to fetch spec.type.doesnotexist

`hasOwnProperty` exists on all javascript types except null and undefined. So I added some tests before I thought of the null and undefined scenarios, and will update to handle those later.

Reply:
Ok. I switched it to loash's `hasIn` which seems to be a polyfill for 'hasOwn`. I like this, cause it serves the need of using lodah to check the key, as well as set the key, so there's less likely to be an odd bug between the two.

I kept it as a reduce for now. I could go back to a for of, that was a little like
```
function getPathArray(input: Entity, field: string) {
  const pathArray = [];
  let currentPathPart = '';

  for (const pathPart of field.split('.')) {
    currentPathPart += pathPart;

    if (lodash.has(input, pathArray.concat(currentPathPart))) {
      pathArray.push(currentPathPart);
      currentPathPart = '';
    } else {
      currentPathPart += '.';
    }
  }

  return pathArray;
}
```

Let me know if you want to be to go back to that approach, but with stepping through the object each cycle to get the value.